src/Microsoft.ML.Core/CommandLine/CmdParser.cs

```diff
@@ -495,6 +500,11 @@ private static ArgumentInfo GetArgumentInfo(Type type, object defaults)
 
                     if (map.ContainsKey(name.ToLowerInvariant()))
                         throw Contracts.Except("Duplicate name '{0}' in argument type '{1}'", name, type.Name);
+                    if (attrName != null)
```

Comment:
>attrName [](start = 24, length = 8)

Same comment here about empty string

Reply:
Not applicable here, since this is attrName, not attr.Name. It's going to be either null or non-empty.